Lib/asyncio/coroutines.py

```diff
@@ -199,12 +198,29 @@ def iscoroutinefunction(func):
 # Prioritize native coroutine check to speed-up
 # asyncio.iscoroutine.
 _COROUTINE_TYPES = (types.CoroutineType, types.GeneratorType,
-                    Coroutine, CoroWrapper)
+                    collections.abc.Coroutine, CoroWrapper)
+_iscoroutine_typecache = {}
```

Comment:
Is it required to be `dict`?
Looks like `set` is enough.

Reply:
Makes sense! Updated both C and Python versions.